tests/unit/retrievers/test_vector.py

```diff
@@ -28,6 +29,26 @@ def test_vector_retriever_initialization(driver):
         mock_verify.assert_called_once()
 
 
+def test_vector_retriever_bad_data_validation(driver):
+    with pytest.raises(ValueError):
+        with patch(
+            "neo4j_genai.retrievers.base.Retriever._verify_version"
+        ) as mock_verify:
+            VectorRetriever(driver=driver, index_name=42)
+            mock_verify.assert_called_once()
```

Comment:
Are we keeping this check for all tests on purpose? Also, do we want to verify version if the provided data is not valid? For instance if the driver is not a valid neo4j driver, we won't even be able to try, right?
Last but not least, I don't understand how this got called looking at the code actually, isn't the version check performed in the super method after the input validation?

Reply:
Good catch!